### PR TITLE
fix for `collections` import deprecation

### PR DESCRIPTION
**Pull request by zacharyburnett**
### Description

Direct import from `collections` is deprecated in Python 3.10. This causes the environment testing in https://github.com/spacetelescope/spacetelescope-env-distribution to fail:
```
/usr/share/miniconda/envs/spacetelescope-env-Linux-py3.10-latest/lib/python3.10/site-packages/stsynphot/commissioning/utils.py:6: in <module>
[29](https://github.com/spacetelescope/spacetelescope-env-distribution/runs/7429139594?check_suite_focus=true#step:11:30)
    from collections import Iterable
[30](https://github.com/spacetelescope/spacetelescope-env-distribution/runs/7429139594?check_suite_focus=true#step:11:31)
E   ImportError: cannot import name 'Iterable' from 'collections' (/usr/share/miniconda/envs/spacetelescope-env-Linux-py3.10-latest/lib/python3.10/collections/__init__.py)
```
https://github.com/spacetelescope/spacetelescope-env-distribution/runs/7429139594?check_suite_focus=true#step:11:31

The solution is to import from `collections.abc` instead.

### CODE REVIEW

File: stsynphot/commissioning/utils.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # STDLIB
 import os
 import sys
-from collections import Iterable
+from collections.abc import Sized
```
Comment (pllim) [inline stsynphot/commissioning/utils.py:6]: Why the change here?
```suggestion
from collections.abc import Iterable
```

**Comment by pllim** [comment]
Good catch but I have a question as stated in the code comment.

Also, why is your test suite importing from the `commissioning` folder?

File: stsynphot/commissioning/utils.py (hunk 1 above)
Comment (zacharyburnett) [reply stsynphot/commissioning/utils.py:6]: from what I can see, this is asking for the `len()` of the object:
https://github.com/spacetelescope/stsynphot_refactor/blob/7c007f9caac36938df5581d9bac2bf4f6b828b38/stsynphot/commissioning/utils.py#L151-L154
so an `Iterable`, which only has to implement `__iter__()` may not implement a `__len__()` method. 
`Sized`, however, requires `__len__()` so would work in all cases.

**Comment by zacharyburnett** [comment]
> Also, why is your test suite importing from the `commissioning` folder?

It looks like this is happening from `pytest` test discovery, as I called `pytest --pyargs stsynphot` to test the package within the environment. So, when it collects tests from `commissioning/tests`, it runs through import of `utils.py`

**Comment by pllim** [comment]
To exclude commissioning folder, please see the `tox.ini` defined in this repo. Your test suite needs to pick this up:

https://github.com/spacetelescope/stsynphot_refactor/blob/272061b0f5db3dcffa41857bab9cc8dd31a22cd0/setup.cfg#L4

File: stsynphot/commissioning/utils.py (hunk 1 above)
Comment (pllim) [reply stsynphot/commissioning/utils.py:6]: OK, that makes sense, I guess. And I don't think it matters in this case anyway, so I'll accept this. Thanks for the clarification!

**Review by pllim — APPROVED** [review]
Thanks!
